bundles/org.openhab.binding.danfossairunit/src/main/java/org/openhab/binding/danfossairunit/internal/discovery/DanfossAirUnitDiscoveryService.java

```diff
@@ -0,0 +1,139 @@
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.net.SocketTimeoutException;
+import java.util.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Discovery service implementation to scan for available air units in the network via broadcast.
+ *
+ * @author Ralf Duckstein - Initial contribution
+ * @author Robert Bach - heavy refactorings
+ */
+@Component(service = DiscoveryService.class, immediate = true)
+@NonNullByDefault
+public class DanfossAirUnitDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int BROADCAST_PORT = 30045;
+    private static final byte[] DISCOVER_SEND = { 0x0c, 0x00, 0x30, 0x00, 0x11, 0x00, 0x12, 0x00, 0x13 };
+    private static final byte[] DISCOVER_RECEIVE = { 0x0d, 0x00, 0x07, 0x00, 0x02, 0x02, 0x00 };
+
+    private Logger logger = LoggerFactory.getLogger(DanfossAirUnitDiscoveryService.class);
+
+    public DanfossAirUnitDiscoveryService() {
+        super(SUPPORTED_THING_TYPES_UIDS, 15, true);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Start Danfoss Air CCM background discovery");
+        scheduler.execute(this::discover);
+    }
+
+    @Override
+    public void startScan() {
+        logger.debug("Start Danfoss Air CCM scan");
+        discover();
+    }
+
+    private synchronized void discover() {
+        logger.debug("Try to discover all Danfoss Air CCM devices");
+
+        try (DatagramSocket socket = new DatagramSocket()) {
+
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = interfaces.nextElement();
+                if (networkInterface == null || networkInterface.isLoopback() || !networkInterface.isUp()) {
+                    continue;
+                }
+                for (InterfaceAddress interfaceAddress : networkInterface.getInterfaceAddresses()) {
+                    if (interfaceAddress == null || interfaceAddress.getBroadcast() == null) {
+                        continue;
+                    }
+                    logger.debug("Sending broadcast on interface {} to discover Danfoss Air CCM device...",
```

Comment:
You don't have to call `toString` it will be called automatically, but only when the logging-level actually applies.